### PR TITLE
gh-113317: Argument Clinic: move linear_format into libclinic

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-113317

### CODE REVIEW

File: Lib/test/test_clinic.py [hunk 1]
```diff
@@ -761,6 +761,19 @@ def test_multiline_substitution(self):
           def
         """, name='bingle\nbungle\n')
 
+    def test_text_before_block_marker(self):
+        regex = "found before '{marker}'"
```
Comment (sobolevn) [inline Lib/test/test_clinic.py]: `re.escape`?

File: Lib/test/test_clinic.py [hunk 2]
```diff
@@ -761,6 +761,19 @@ def test_multiline_substitution(self):
...
+        with self.assertRaisesRegex(clinic.ClinicError, regex):
+            libclinic.linear_format("no text before marker for you! {marker}",
+                                    marker="not allowed!")
+
+    def test_text_after_block_marker(self):
+        regex = "found after '{marker}'"
```
Comment (sobolevn) [inline Lib/test/test_clinic.py]: The same here.

File: Lib/test/test_clinic.py (hunk 1 above)
Comment (erlend-aasland) [reply Lib/test/test_clinic.py]: That's a good idea.

**Review by sobolevn — APPROVED** [review]
👍

**Pull request merged**